### PR TITLE
feat(datasource/docker): use schema and support annotations

### PR DESCRIPTION
**Pull request by viceice**
## Changes
- Use zod schema to validate
- Support manifest annotations as label replacement
- Support helm configurations to get source url

**TODO**:
- [x] fix code coverage
- [x] real repo tests: 
  - [ ] https://github.com/renovate-reproductions/23001-oci-helm-chart-metadata/pull/2
  - [ ] https://github.com/renovate-reproductions/23001-oci-helm-chart-metadata/pull/4
  - [ ] https://github.com/renovate-reproductions/24144-gitea-changelogs/pull/5

## Context
- #23001

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
Will do the real repo runs later 🤗

**Review by secustor — COMMENTED** [review]
LGTM so far

**Comment by viceice** [comment]
I don't know any docker images or oci helm charts which already have those new annotations. so i could only test the source url findings.

**Comment by viceice** [comment]
normal docker images still work: https://github.com/renovate-reproductions/23001-oci-helm-chart-metadata/pull/4

so please review 😉

**Comment by viceice** [comment]
changelogs for oci helm charts are working with this PR 🙃 

https://github.com/renovate-reproductions/24144-gitea-changelogs/pull/5

File: lib/modules/datasource/docker/index.ts [hunk 1]
```diff
@@ -196,105 +201,130 @@ export class DockerDatasource extends Datasource {
...
+            dockerRepository,
+            manifest.manifests[0].digest
+          );
+        } else {
+          logger.debug(
+            { manifest },
+            'Invalid manifest list with no manifests - returning'
+          );
+          return null;
+        }
```
Comment (secustor) [inline lib/modules/datasource/docker/index.ts:323]:
```suggestion
        if (!manifest.manifests.length) {
            logger.debug(
            { manifest },
            'Invalid manifest list with no manifests - returning'
          );
          return null;
        }
        logger.trace(
          { registry, dockerRepository, tag },
          'Found manifest list, using first image'
        );
        return this.getManifest(
          registry,
          dockerRepository,
          manifest.manifests[0].digest
        );
```

**Review by secustor — COMMENTED** [review]
Else LGTM, tough I can not confirm either if the media types are correct

**Comment by viceice** [comment]
> Else LGTM, tough I can not confirm either if the media types are correct

The media types are the same as before and i validated them against existing objects and docs

**Review by secustor — APPROVED** [review]

**Pull request merged**